boot/seal.go

```diff
@@ -407,3 +534,19 @@ func isResealNeeded(pbc predictableBootChains, rootdir string, expectReseal bool
 	}
 	return true, c + 1, nil
 }
+
+func isFallbackResealNeeded(pbc predictableBootChains, rootdir string, expectReseal bool) (ok bool, nextCount int, err error) {
```

Comment:
this can probably be one helper somehow  called twice with the right arguments, instead of two full separate helpers

Reply:
Using a separate file for recovery boot chains allows us to use the same helper for both cases (but passing the full file path instead of just the rootdir).